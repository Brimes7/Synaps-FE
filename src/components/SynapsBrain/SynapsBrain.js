import React from "react";
import PropTypes from "prop-types";
import { SvgContainer } from "../SvgContainer/SvgContainer.js";

/**
 *  Synaps Brain SVG
 *
 * @description Displays a svg of the synaps brain adjustable styles.
 *
 *  @component
 *  @example
 *  return (
 *    <SynapsBrain
 *       svg={svg}
 *       zIndex={ 1 }
 *       position={ 'absolute' }
 *       backgroundColor={ theme.primaryColor }
 *       color={ '#164172' }
 *       opacity={ 1 }
 *       strokeColor={ theme.primaryColor }
 *       viewBox={ '225 25 400 400' }
 *    />
 *    )
 */
export const SynapsBrain = ( props ) => {
  
  return <h1>Synaps Brain</h1>;
};

SynapsBrain.propTypes = {
  position: PropTypes.string,
  left: PropTypes.string,
  top: PropTypes.string,
  backgroundColor: PropTypes.string,
  opacity: PropTypes.number,
  color: PropTypes.string,
  strokeColor: PropTypes.string,
  stokeWidth: PropTypes.string,
  viewBox: PropTypes.string,
  zIndex: PropTypes.number,
  width: PropTypes.string,
  height: PropTypes.string,
};
