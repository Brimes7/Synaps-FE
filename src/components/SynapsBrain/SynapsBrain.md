```js
import {useThemeContext} from './UseThemeContext.js';
import {ContainerDiv} from '../Container/ContainerDiv.js';
<ContainerDiv position="relative" width={"300px"} height="300px">
    <SynapsBrain
              zIndex={ 1 }
              position={ 'absolute' }
              backgroundColor={ useThemeContext.primaryColor }
              color={ '#164172' } opacity={ 1 }
              strokeColor={ useThemeContext.primaryColor }
              viewBox={ '225 25 400 400' }
            />
    <h2 style={{zIndex: 3, color: "white"}}>Text</h2>
</ContainerDiv>
```