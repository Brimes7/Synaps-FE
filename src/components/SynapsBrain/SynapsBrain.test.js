import React from 'react';
import { customRender } from '../../utilities/test-utils.js';

import { SynapsBrain } from './SynapsBrain.js';

test( 'Synaps Brain renders correctly', () => {
  const { container } = customRender( <SynapsBrain/> );
  expect( container ).toMatchSnapshot();
} );