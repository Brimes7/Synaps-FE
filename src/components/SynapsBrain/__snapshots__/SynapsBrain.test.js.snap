// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCardDeck render correctly 1`] = `
.c0 {
  position: block;
  left: 0;
  top: 0;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  height: -webkit-min-content;
  height: -moz-min-content;
  height: min-content;
}

.c1 {
  position: relative;
}

.c2 {
  z-index: 5;
  background: #FFFFFF;
}

.c2 > *::before,
.c2 *::after,
.c2 * {
  z-index: 5;
  opacity: 1;
  fill: #0d2545;
}

.c2 > g {
  z-index: 5;
  stroke: #FFFFFF;
  stroke-width: 1px;
}

<div>
  <div
    class="c0"
    data-testid="positional-container"
  >
    <div
      class="c1"
      data-testid="relative-container"
    >
      <svg
        class="c2"
        data-testid="synaps-brain"
        theme="[object Object]"
      >
        Frame.svg
      </svg>
    </div>
  </div>
</div>
`;

exports[`Synaps Brain renders correctly 1`] = `
.c0 {
  position: block;
  left: 0;
  top: 0;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  height: -webkit-min-content;
  height: -moz-min-content;
  height: min-content;
}

.c1 {
  position: relative;
}

.c2 {
  z-index: 5;
  background: #FFFFFF;
}

.c2 > *::before,
.c2 *::after,
.c2 * {
  z-index: 5;
  opacity: 1;
  fill: #0d2545;
}

.c2 > g {
  z-index: 5;
  stroke: #FFFFFF;
  stroke-width: 1px;
}

<div>
  <div
    class="c0"
    data-testid="positional-container"
  >
    <div
      class="c1"
      data-testid="relative-container"
    >
      <svg
        class="c2"
        data-testid="synaps-brain"
        theme="[object Object]"
      >
        Frame.svg
      </svg>
    </div>
  </div>
</div>
`;
